File: Builder/Golang/ExcelReportBuilder.go
package main

type ExcelReportBuilder struct {
	report *Report
}

func (excelReportBuilder *ExcelReportBuilder) buildContent() {
	if excelReportBuilder.report == nil {
		excelReportBuilder.report = &Report{}
	}

	excelReportBuilder.report.setContent("Tables")
}

func (excelReportBuilder *ExcelReportBuilder) buildFile() {
	if excelReportBuilder.report == nil {
		excelReportBuilder.report = &Report{}
	}

	excelReportBuilder.report.setFilename("Report.xlsx") 
}

func (excelReportBuilder *ExcelReportBuilder) getResult() *Report {
	return excelReportBuilder.report
}

File: Builder/Dart/ExcelReportBuilder.dart
import 'IReportBuilder.dart';
import 'Report.dart';

class ExcelReportBuilder implements IReportBuilder {
  Report? _report;

  @override
  void buildContent() {
    if (_report == null) {
      _report = Report();
    }

    _report?.setContent("Tables");
  }

  @override
  void buildFile() {
    if (_report == null) {
      _report = Report();
    }

    _report?.setFilename("Report.xlsx");
  }

  @override
  Report? getResult() {
    return _report;
  }
}
